### PR TITLE
Feature/37 ocr

### LINKED ISSUE #37: Add basic version of text component on back
DoD:
- [ ] created server that get basic with e.g. one param request and returns correct data

### LINKED ISSUE #42: Add more params to text component on back
DoD:
- [x] decide what text params will be added
- [ ] add more text params

params:
- contain any text
- contain specific text 
- min and max length

### PR DESCRIPTION
**Pull request by ostatni5**
close #37 
close #42 

Download tesseract:
https://digi.bib.uni-mannheim.de/tesseract/tesseract-ocr-w64-setup-v4.1.0.20190314.exe

Install elixir 

Install dependencies in the project folder
mix local.hex
mix deps.get

Run app
iex -S mix

Run tests
mix test

### CODE REVIEW

File: elixir_server/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+def deps do
+  [
+    {:elixir_server, "~> 0.1.0"}
+  ]
+end
+```
+
+Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc)
+and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
+be found at <https://hexdocs.pm/elixir_server>.
```
Comment (michal367) [inline elixir_server/README.md:20]: Does this part concern us?

File: elixir_server/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Info
+
+Download tesseract:
+<https://digi.bib.uni-mannheim.de/tesseract/tesseract-ocr-w64-setup-v4.1.0.20190314.exe>
```
Comment (michal367) [inline elixir_server/README.md:25]: Also works on tesseract 5.0
Maybe add link to https://tesseract-ocr.github.io/tessdoc/Installation.html

File: elixir_server/lib/elixir_server/ocr.ex [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {name, param} = option
+
+    case name do
+      "hasText" ->
+        &hasText/1
+
+      "containsText" ->
+        &containsText(String.downcase(param), String.downcase(&1))
+
+      "minLenght" ->
```
Comment (michal367) [inline elixir_server/lib/elixir_server/ocr.ex:43]: lengTH!!!

File: elixir_server/lib/elixir_server/ocr.ex [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        &minLength(param, &1)
+
+      "maxLenght" ->
```
Comment (michal367) [inline elixir_server/lib/elixir_server/ocr.ex:46]: lengTH!!!

File: elixir_server/mix.exs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  # Run "mix help deps" to learn about dependencies.
+  defp deps do
+    [
+      {:plug, "~> 1.5"},
+      {:poison, "~> 3.1"},
+      {:plug_cowboy, "~> 1.0"},
+      {:tesseract_ocr, "~> 0.1.5"}
+      # {:dep_from_hexpm, "~> 0.3.0"},
+      # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"}
```
Comment (michal367) [inline elixir_server/mix.exs:30]: remove these comments.
In fact, you can delete all comments in this file.

File: elixir_server/test/elixir_server_test.exs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+  @options ElixirServer.Router.init([])
+
+  @rootDir File.cwd!()
+
+  @data %{
+    "options" => %{},
+    "paths" => [
+      "#{@rootDir}/test/resources/not_text.png",
+      "#{@rootDir}/test/resources/not_text.png",
```
Comment (michal367) [inline elixir_server/test/elixir_server_test.exs:20]: Why almost everywhere in this file are doubled paths?

File: elixir_server/lib/elixir_server/ocr.ex [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+  def maxLength(number, outputOcr) do
+    compareLength(&Kernel.<=/2, number, outputOcr)
+  end
+
+  def parseOption(option) do
+    {name, param} = option
+
+    case name do
+      "hasText" ->
```
Comment (michal367) [inline elixir_server/lib/elixir_server/ocr.ex:37]: do we want this param? Shouldn't it be that the default behavior (without params) of this program is to send back image paths with text?

File: elixir_server/lib/elixir_server/ocr.ex [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+      "containsText" ->
+        &containsText(String.downcase(param), String.downcase(&1))
+
+      "minLenght" ->
+        &minLength(param, &1)
+
+      "maxLenght" ->
+        &maxLength(param, &1)
+
```
Comment (michal367) [inline elixir_server/lib/elixir_server/ocr.ex:48]: Where is case-sensitive parameter that affects the containsText?

File: elixir_server/lib/elixir_server.ex [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+  ## Examples
+
+      iex> ElixirServer.hello()
+      :world
+
+  """
+  def hello do
+    :world
+  end
```
Comment (michal367) [inline elixir_server/lib/elixir_server.ex:17]: I think we don't need this

File: server/tests/traverse.test.ts [hunk 10]
```diff
@@ -1,5 +1,5 @@
 import supertest from 'supertest';
-import traverse from '../src/modules/traverseModule/traverse';
+import traverse from '../src/modules/traverseModule/main';
```
Comment (michal367) [inline server/tests/traverse.test.ts:2]: Change also directory name to `mainModule`

**Review by michal367 — CHANGES_REQUESTED** [review]

File: elixir_server/README.md (hunk 2 above)
Comment (ostatni5) [reply elixir_server/README.md:25]: API is fully compatible with 4.x.x we don't know if everything will be working with 5.0

File: elixir_server/README.md (hunk 1 above)
Comment (ostatni5) [reply elixir_server/README.md:20]: Could be helpful but I don't know.

File: elixir_server/test/elixir_server_test.exs (hunk 6 above)
Comment (ostatni5) [reply elixir_server/test/elixir_server_test.exs:20]: Just to test if it is working for same picture two times

File: elixir_server/lib/elixir_server/ocr.ex (hunk 7 above)
Comment (ostatni5) [reply elixir_server/lib/elixir_server/ocr.ex:37]: added parsing true/false

File: elixir_server/mix.exs (hunk 5 above)
Comment (ostatni5) [reply elixir_server/mix.exs:30]: Nope if some want to add something they can be helpful

**Review by michal367 — APPROVED** [review]

**Pull request merged**